src/shogun/machine/GLM.cpp

```diff
@@ -0,0 +1,408 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Tej Sukhatme
+ */
+
+#include <shogun/features/DotFeatures.h>
+#include <shogun/labels/Labels.h>
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/machine/LinearMachine.h>
+#include <shogun/machine/GLM.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/mathematics/NormalDistribution.h>
+#include <shogun/mathematics/RandomNamespace.h>
+#include <shogun/optimization/ElasticNetPenalty.h>
+#include <shogun/optimization/SGDMinimizer.h>
+#include <shogun/optimization/GradientDescendUpdater.h>
+#include <shogun/optimization/ConstLearningRate.h>
+// #include <utility>
+#include <cmath>
+
+using namespace shogun;
+
+GLM::GLM(GLM_DISTRIBUTION distr, float64_t alpha, float64_t lambda, float64_t learning_rate, int32_t max_iterations, float64_t tolerance, float64_t eta): IterativeMachine<LinearMachine>()
+{
+	init();
+	distribution=distr;
+	m_alpha=alpha;
+	m_lambda=lambda;
+	m_learning_rate=learning_rate;
+	m_max_iterations=max_iterations;
+	m_tolerance=tolerance;
+	m_eta=eta;
+}
+
+void GLM::init()
+{
+	bias = 0;
+
+	distribution= POISSON;
```

Comment:
you can initialise in the declaration of the variables, then you don't need the init method and just call the std ctor in all other ctors

Reply:
Okay, I will remove this.